### DISCUSSION TITLE
Log Analytics throttling handling mechanism

### DISCUSSION THREAD

**Discussion by johnnyfeldman** [grafana/grafana#49267] [Data Sources]
Hi All,
When using Log Analytics data source, it actually uses a specific SPN to run the query. All users of the dashboard are sharing this SPN.
Azure has throttling policies causing the SPN to get throttled and thus causing all the users to experience timeouts and not working dashboards. Even with most efficient queries, there is no way to avoid throttling since number of concurrent queries are part of the policy, meaning that multiple users working with the same dashboard will eventually get throttled. 
I want to suggest a feature where a dashboard developer can provide multiple accounts and the data source will pick a different one each time it runs a query and also will detect response codes that indicates throttling had occurred to the account and switch automatically to another account.

Thank you,
Johnny.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82994. Feel free to continue the discussion around this there. Thank you!
